sdk/core/System.ClientModel/tests/Convenience/ClientResultTests.cs

```diff
@@ -127,6 +127,18 @@ public void CanCreateDerivedClientResultOfT()
         Assert.AreEqual(response.Status, result.GetRawResponse().Status);
     }
 
+    [Test]
```

Comment:
could you add a test for ClientResponse<string?> and possibly ClientResponse<int?>, i.e. we want to support optional Ts too and then the cast should return null/default